server/workflowarchive/archived_workflow_server.go

```diff
@@ -188,3 +191,25 @@ func (w *archivedWorkflowServer) ListArchivedWorkflowLabelValues(ctx context.Con
 	}
 	return labels, err
 }
+
+func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req *workflowarchivepkg.RetryArchivedWorkflowRequest) (*wfv1.Workflow, error) {
+	wfClient := auth.GetWfClient(ctx)
+	kubeClient := auth.GetKubeClient(ctx)
+
+	wf, err := w.wfArchive.GetWorkflow(req.Uid)
+	if err != nil {
+		return nil, err
+	}
+
+	err = w.hydrator.Hydrate(wf)
```

Comment:
this is a no-op - workflows in the archive will always be hydrated

Reply:
In this case, I think we can remove the hydrator altogether from the workflow archive since it was only added for this call.